packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -22,6 +22,23 @@ bool isDartFile(FileSystemEntity entry) => entry is File && entry.path.endsWith(
 bool isDartTestFile(FileSystemEntity entry) => entry is File && entry.path.endsWith('_test.dart');
 bool isDartBenchmarkFile(FileSystemEntity entry) => entry is File && entry.path.endsWith('_bench.dart');
 
+RegExp _testFileParser = new RegExp(r'^(.+)_test(\.dart)$');
+
+void _addDriverTest(FileSystemEntity entry, List<String> dartFiles) {
+  if (isDartTestFile(entry)) {
+    final String testFileName = entry.path;
+    dartFiles.add(testFileName);
+    Match groups = _testFileParser.firstMatch(testFileName);
+    assert(groups.groupCount == 2);
+    final String hostFileName = '${groups[1]}${groups[2]}';
+    File hostFile = new File(hostFileName);
+    if (hostFile.existsSync()) {
```

Comment:
Perhaps even `assert(hostFile.existsSync)`? This way if the convention changes (I have an issue assigned to me to change it) the analyzer breaks rather than ignore missing files.